packages/oauth/src/index.ts

```diff
@@ -7,11 +7,29 @@ export interface OAuthProvider {
 		providerUser: Record<string, any>;
 		[data: string]: any;
 	}>;
-	getAuthorizationUrl: () => string;
+	getAuthorizationUrl: (state?: string) => [url: string, state: string | undefined];
+}
+
+export interface OAuthConfig {
+	clientId: string;
+	clientSecret: string;
+	scope?: string[];
+	state?: boolean;
```

Comment:
I also prefer to have state configurable in `OAuthConfig` but can’t for the life of me make it configurable. It might better if state is configurable in `getAuthorizationUrl()`. Refer to other comment.